src/ir/irconv/irconv_test.go

```diff
@@ -313,3 +318,59 @@ func TestInterpretString(t *testing.T) {
 		runTest(t, test.raw, '"', test.doubleQuote)
 	}
 }
+
+func TestCurlyBraceArrayDimFetch(t *testing.T) {
+	tests := []struct {
+		code       string
+		curlyBrace bool
+	}{
+		{
+			code:       `$arr{'x'};`,
+			curlyBrace: true,
+		},
+		{
+			code:       `$arr['x'];`,
+			curlyBrace: false,
+		},
+		{
+			code:       `$arr  [  'x'   ]   ;`,
+			curlyBrace: false,
+		},
+		{
+			code:       `$arr  {  'x'   }   ;`,
+			curlyBrace: true,
+		},
+		{
+			code:       `$arr{  'x'   };`,
+			curlyBrace: true,
+		},
+		{
+			code:       `$arr[  'x'   ];`,
+			curlyBrace: false,
+		},
+		{
+			code:       `$arr{'x'   };`,
+			curlyBrace: true,
+		},
+		{
+			code:       `$arr['x'   ];`,
+			curlyBrace: false,
+		},
+	}
+
+	for _, test := range tests {
+		p := php7.NewParser([]byte(`<?php ` + test.code))
```

Comment:
Is it possible to use parseutil function for parsing here?
It's painful to use the parser directly in different places as we're going to migrate soon.